### PR TITLE
Handle aliases in chart dependencies

### PR DESCRIPTION
**Pull request by hardys**
It's valid to have multiple dependencies with alternate aliases which currently are removed on chart generation.

One way to resolve this is just to operate directly on the dependencies list instead of generating a map keyed by name, which is less efficient but avoids the problem of removing valid dependencies that use aliases.

Fixes: #18

### CODE REVIEW

File: pkg/charts/dependencies.go [hunk 1]
```diff
@@ -250,35 +250,28 @@ func UpdateHelmMetadataWithDependencies(fs billy.Filesystem, mainHelmChartPath s
...
-	}
+
 	// Update the Repository for each dependency
 	for dependencyName := range dependencyMap {
-		d, ok := helmDependencyMap[dependencyName]
-		if !ok {
+		found := false
+		for _, d := range chart.Metadata.Dependencies {
+			if d.Name == dependencyName {
+				d.Repository = fmt.Sprintf("file://./charts/%s", dependencyName)
```
Comment (rohitsakala) [inline pkg/charts/dependencies.go:259]: Did we miss doing d.Version =. ""  ? here wantedly ?

Comment (hardys) [reply pkg/charts/dependencies.go:259]: Good point - I had been looking at https://github.com/rancher/charts-build-scripts/issues/68 and was thinking that this remapping of repositories probably shouldn't mess with any existing version.

But that is an separate issue so I'll reinstate the version-removal logic here, and we can perhaps discuss the version issue separately via a follow-up PR.

Comment (rohitsakala) [reply pkg/charts/dependencies.go:259]: Okay thanks. Please update the PR and make a new PR

Comment (hardys) [reply pkg/charts/dependencies.go:259]: Done!

**Review by rohitsakala — APPROVED** [review]

**Pull request merged**